server/src/main/java/org/elasticsearch/search/aggregations/bucket/histogram/AutoDateHistogramAggregationBuilder.java

```diff
@@ -92,6 +103,22 @@ public static AutoDateHistogramAggregationBuilder parse(String aggregationName,
 
     private int numBuckets = 10;
 
+    private String minimumIntervalExpression;
+
+    public String getMinimumIntervalExpression() {
+        return minimumIntervalExpression;
+    }
+
+    public AutoDateHistogramAggregationBuilder setMinimumIntervalExpression(String minimumIntervalExpression) {
+        if (minimumIntervalExpression != null && !allowedIntervals.contains(minimumIntervalExpression)) {
```

Comment:
```suggestion
        if (minimumIntervalExpression != null && allowedIntervals.contains(minimumIntervalExpression) == false) {
```

We try to avoid `!(expr)` since it's easy to miss the negation :)